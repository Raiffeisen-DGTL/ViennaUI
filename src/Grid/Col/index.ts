export { Col, type ColProps } from './Col';
