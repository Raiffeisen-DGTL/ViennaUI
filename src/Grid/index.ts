export { Grid } from './Grid';
export {
    type ColProps as GridColProps,
    type BasedSize as GridBasedSize,
    type ColumnOffset as GridColumnOffset,
    type ColumnOrder as GridColumnOrder,
    type ColumnSize as GridColumnSize,
    type OffsetOption as GridOffsetOption,
    type OrderOption as GridOrderOption,
    type SizeOption as GridSizeOption,
    type Sizes as GridSizes,
} from './Col';

export { type RowProps as GridRowProps } from './Row';
