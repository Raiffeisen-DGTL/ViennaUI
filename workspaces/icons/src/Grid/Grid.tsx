import React, { SVGAttributes } from 'react';

export interface GridProps extends SVGAttributes<SVGElement> {
    [key: string]: any;
    size?: 'xs' | 's' | 'm' | 'l' | 'xl' | number;
    color?: string;
}

const sizes = { xs: 12, s: 16, m: 20, l: 24, xl: 32 };
export const Grid: React.FC<GridProps> = (props): JSX.Element => {
    const { color = 'currentColor', size = 'm', ...attrs } = props;
    const d = sizes[size] || size;

    return (
        <svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24' width={d} height={d} fill={color} {...attrs}>
            <defs />
            <path
                fillRule='evenodd'
                d='M4 3a1 1 0 00-1 1v6a1 1 0 001 1h6a1 1 0 001-1V4a1 1 0 00-1-1H4zm1 6V5h4v4H5zm-1 4a1 1 0 00-1 1v6a1 1 0 001 1h6a1 1 0 001-1v-6a1 1 0 00-1-1H4zm1 6v-4h4v4H5zm8-15a1 1 0 011-1h6a1 1 0 011 1v6a1 1 0 01-1 1h-6a1 1 0 01-1-1V4zm2 1v4h4V5h-4zm-1 8a1 1 0 00-1 1v6a1 1 0 001 1h6a1 1 0 001-1v-6a1 1 0 00-1-1h-6zm1 6v-4h4v4h-4z'
                clipRule='evenodd'
            />
        </svg>
    );
};

Grid.defaultProps = {
    size: 'm',
    color: 'currentColor',
};

Grid.displayName = 'Grid';
export default Grid;
