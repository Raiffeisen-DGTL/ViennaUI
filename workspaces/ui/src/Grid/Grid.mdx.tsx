import styled from 'styled-components';

export const Demo = styled.div`
    line-height: 2rem;
    text-align: center;
    background: #018ffe;
    border: 1px solid #fff;
    border-radius: 2px;
    color: #fff;
    font-size: 14px;
    height: 100%;
    box-sizing: border-box;
`;

export const TallDemo = styled(Demo)`
    height: 100px;
`;
