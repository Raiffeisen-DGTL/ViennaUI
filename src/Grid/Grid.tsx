import { Col } from './Col';
import { Row } from './Row';

export const Grid = { Col, Row };
