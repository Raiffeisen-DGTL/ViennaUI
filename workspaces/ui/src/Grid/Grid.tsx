import * as Col from './Col';
import * as Row from './Row';

export const Grid: { Col: React.FC<Col.ColProps>; Row: React.FC<Row.RowProps> } = {
    Col: Col.Col,
    Row: Row.Row,
};

export default Grid;
