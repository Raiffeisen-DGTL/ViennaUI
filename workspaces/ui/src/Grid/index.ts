import * as _Col from './Col';
import * as _Row from './Row';

export const Grid: { Col: React.FC<_Col.ColProps>; Row: React.FC<_Row.RowProps> } = {
    Col: _Col.Col,
    Row: _Row.Row,
};
