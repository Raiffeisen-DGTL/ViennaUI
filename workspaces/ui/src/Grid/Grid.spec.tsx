import React from 'react';
import Grid from './Grid';

test('Grid', () => {
    const snap = snapshot.render(
        <>
            <Grid.Row>
                <Grid.Col size={1}>1</Grid.Col>
                <Grid.Col size={1}>1</Grid.Col>
                <Grid.Col size={1}>1</Grid.Col>
                <Grid.Col size={1}>1</Grid.Col>
                <Grid.Col size={1}>1</Grid.Col>
                <Grid.Col size={1}>1</Grid.Col>
                <Grid.Col size={1}>1</Grid.Col>
                <Grid.Col size={1}>1</Grid.Col>
                <Grid.Col size={1}>1</Grid.Col>
                <Grid.Col size={1}>1</Grid.Col>
                <Grid.Col size={1}>1</Grid.Col>
                <Grid.Col size={1}>1</Grid.Col>
            </Grid.Row>
            <Grid.Row>
                <Grid.Col size={6}>One of two columns</Grid.Col>
                <Grid.Col size={6}>One of two columns</Grid.Col>
            </Grid.Row>
            <Grid.Row>
                <Grid.Col size={4}>One of three columns</Grid.Col>
                <Grid.Col size={4}>One of three columns</Grid.Col>
                <Grid.Col size={4}>One of three columns</Grid.Col>
            </Grid.Row>
            <Grid.Row>
                <Grid.Col size={3}>One of four columns</Grid.Col>
                <Grid.Col size={3}>One of four columns</Grid.Col>
                <Grid.Col size={3}>One of four columns</Grid.Col>
                <Grid.Col size={3}>One of four columns</Grid.Col>
            </Grid.Row>
        </>
    );
    expect(snap).toMatchSnapshot();
});

test('Grid w/ columnGap', () => {
    const snap = snapshot.render(
        <>
            <Grid.Row>
                <Grid.Col size={1}>24</Grid.Col>
                <Grid.Col size={1}>24</Grid.Col>
                <Grid.Col size={1}>24</Grid.Col>
                <Grid.Col size={1}>24</Grid.Col>
                <Grid.Col size={1}>24</Grid.Col>
                <Grid.Col size={1}>24</Grid.Col>
                <Grid.Col size={1}>24</Grid.Col>
                <Grid.Col size={1}>24</Grid.Col>
                <Grid.Col size={1}>24</Grid.Col>
                <Grid.Col size={1}>24</Grid.Col>
                <Grid.Col size={1}>24</Grid.Col>
                <Grid.Col size={1}>24</Grid.Col>
            </Grid.Row>
            <Grid.Row columnGap='12px'>
                <Grid.Col size={1}>12</Grid.Col>
                <Grid.Col size={1}>12</Grid.Col>
                <Grid.Col size={1}>12</Grid.Col>
                <Grid.Col size={1}>12</Grid.Col>
                <Grid.Col size={1}>12</Grid.Col>
                <Grid.Col size={1}>12</Grid.Col>
                <Grid.Col size={1}>12</Grid.Col>
                <Grid.Col size={1}>12</Grid.Col>
                <Grid.Col size={1}>12</Grid.Col>
                <Grid.Col size={1}>12</Grid.Col>
                <Grid.Col size={1}>12</Grid.Col>
                <Grid.Col size={1}>12</Grid.Col>
            </Grid.Row>
            <Grid.Row columnGap='5px'>
                <Grid.Col size={1}>5</Grid.Col>
                <Grid.Col size={1}>5</Grid.Col>
                <Grid.Col size={1}>5</Grid.Col>
                <Grid.Col size={1}>5</Grid.Col>
                <Grid.Col size={1}>5</Grid.Col>
                <Grid.Col size={1}>5</Grid.Col>
                <Grid.Col size={1}>5</Grid.Col>
                <Grid.Col size={1}>5</Grid.Col>
                <Grid.Col size={1}>5</Grid.Col>
                <Grid.Col size={1}>5</Grid.Col>
                <Grid.Col size={1}>5</Grid.Col>
                <Grid.Col size={1}>5</Grid.Col>
            </Grid.Row>
            <Grid.Row columnGap='0'>
                <Grid.Col size={1}>0</Grid.Col>
                <Grid.Col size={1}>0</Grid.Col>
                <Grid.Col size={1}>0</Grid.Col>
                <Grid.Col size={1}>0</Grid.Col>
                <Grid.Col size={1}>0</Grid.Col>
                <Grid.Col size={1}>0</Grid.Col>
                <Grid.Col size={1}>0</Grid.Col>
                <Grid.Col size={1}>0</Grid.Col>
                <Grid.Col size={1}>0</Grid.Col>
                <Grid.Col size={1}>0</Grid.Col>
                <Grid.Col size={1}>0</Grid.Col>
                <Grid.Col size={1}>0</Grid.Col>
            </Grid.Row>
        </>
    );
    expect(snap).toMatchSnapshot();
});

test('Grid w/ rowGap', () => {
    const snap = snapshot.render(
        <>
            <Grid.Row rowGap='5px'>
                <Grid.Col size={12}>5px row gap</Grid.Col>
            </Grid.Row>
            <Grid.Row rowGap='10px'>
                <Grid.Col size={12}>10px row gap</Grid.Col>
            </Grid.Row>
            <Grid.Row>
                <Grid.Col size={12}>No row gap</Grid.Col>
            </Grid.Row>
            <Grid.Row>
                <Grid.Col size={12}>No row gap</Grid.Col>
            </Grid.Row>
        </>
    );
    expect(snap).toMatchSnapshot();
});

test('Grid w/ offset', () => {
    const snap = snapshot.render(
        <>
            <Grid.Row>
                <Grid.Col offset={11} size={1}>
                    1
                </Grid.Col>
                <Grid.Col offset={10} size={2}>
                    2
                </Grid.Col>
                <Grid.Col offset={9} size={3}>
                    3
                </Grid.Col>
                <Grid.Col offset={8} size={4}>
                    4
                </Grid.Col>
                <Grid.Col offset={7} size={5}>
                    5
                </Grid.Col>
                <Grid.Col offset={6} size={6}>
                    6
                </Grid.Col>
                <Grid.Col offset={5} size={7}>
                    7
                </Grid.Col>
                <Grid.Col offset={4} size={8}>
                    8
                </Grid.Col>
                <Grid.Col offset={3} size={9}>
                    9
                </Grid.Col>
                <Grid.Col offset={2} size={10}>
                    10
                </Grid.Col>
                <Grid.Col offset={1} size={11}>
                    11
                </Grid.Col>
            </Grid.Row>
            <Grid.Row>
                <Grid.Col offset={3} size={3}>
                    offset: 3
                </Grid.Col>
                <Grid.Col offset={2} size={3}>
                    offset: 2
                </Grid.Col>
            </Grid.Row>
            <Grid.Row>
                <Grid.Col size={3}>size 3</Grid.Col>
                <Grid.Col offset={3} size={2}>
                    size 2
                </Grid.Col>
                <Grid.Col offset={3} size={1}>
                    1
                </Grid.Col>
            </Grid.Row>
        </>
    );
    expect(snap).toMatchSnapshot();
});

test('Grid adaptive', () => {
    const sizes: any = [
        { xs: 12, s: 10, m: 8, l: 6, xl: 4 },
        { xs: 12, s: 2, m: 4, l: 6, xl: 8 },
        { xs: 12, s: 2, m: 4, l: 6, xl: 8 },
        { xs: 12, s: 2, m: 4, l: 6, xl: 8 },
        { xs: 12, s: 2, m: 4, l: 2, xl: 3 },
    ];

    const snap = snapshot.render(
        <>
            <Grid.Row>
                <Grid.Col size={sizes[0]}>1</Grid.Col>
                <Grid.Col size={sizes[1]}>2</Grid.Col>
            </Grid.Row>
            <Grid.Row>
                <Grid.Col size={sizes[2]} offset={sizes[3]}>
                    1
                </Grid.Col>
                <Grid.Col size={sizes[4]}>2</Grid.Col>
            </Grid.Row>
        </>
    );
    expect(snap).toMatchSnapshot();
});

test('Grid w align', () => {
    const snap = snapshot.render(
        <>
            <Grid.Row align='left'>
                <Grid.Col size={2}>1</Grid.Col>
                <Grid.Col size={2}>1</Grid.Col>
                <Grid.Col size={2}>1</Grid.Col>
            </Grid.Row>
            <Grid.Row align='center'>
                <Grid.Col size={2}>1</Grid.Col>
                <Grid.Col size={2}>1</Grid.Col>
                <Grid.Col size={2}>1</Grid.Col>
            </Grid.Row>
            <Grid.Row align='right'>
                <Grid.Col size={2}>1</Grid.Col>
                <Grid.Col size={2}>1</Grid.Col>
                <Grid.Col size={2}>1</Grid.Col>
            </Grid.Row>
            <Grid.Row align='around'>
                <Grid.Col size={2}>1</Grid.Col>
                <Grid.Col size={2}>1</Grid.Col>
                <Grid.Col size={2}>1</Grid.Col>
            </Grid.Row>
            <Grid.Row align='between'>
                <Grid.Col size={2}>1</Grid.Col>
                <Grid.Col size={2}>1</Grid.Col>
                <Grid.Col size={2}>1</Grid.Col>
            </Grid.Row>
            <Grid.Row align='stretch'>
                <Grid.Col>1</Grid.Col>
                <Grid.Col>1</Grid.Col>
                <Grid.Col>1</Grid.Col>
            </Grid.Row>
        </>
    );
    expect(snap).toMatchSnapshot();
});

test('Grid w valign', () => {
    const snap = snapshot.render(
        <>
            <Grid.Row valign='top'>
                <Grid.Col size={6}>Tall column</Grid.Col>
                <Grid.Col size={6}>top</Grid.Col>
            </Grid.Row>
            <Grid.Row valign='middle'>
                <Grid.Col size={6}>Tall column</Grid.Col>
                <Grid.Col size={6}>middle</Grid.Col>
            </Grid.Row>
            <Grid.Row valign='bottom'>
                <Grid.Col size={6}>Tall column</Grid.Col>
                <Grid.Col size={6}>bottom</Grid.Col>
            </Grid.Row>
            <Grid.Row>
                <Grid.Col size={6}>Tall column</Grid.Col>
                <Grid.Col size={6}>bottom</Grid.Col>
            </Grid.Row>
        </>
    );
    expect(snap).toMatchSnapshot();
});
