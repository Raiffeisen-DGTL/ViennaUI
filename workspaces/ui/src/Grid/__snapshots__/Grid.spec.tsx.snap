// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid 1`] = `
Array [
  <div
    class="Row-sc-19nxm1v-0 cwCUyd"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      1
    </div>
  </div>,
  <div
    class="Row-sc-19nxm1v-0 cwCUyd"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 hiyiVh"
      offset="0"
      size="6"
    >
      One of two columns
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 hiyiVh"
      offset="0"
      size="6"
    >
      One of two columns
    </div>
  </div>,
  <div
    class="Row-sc-19nxm1v-0 cwCUyd"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 btUUfi"
      offset="0"
      size="4"
    >
      One of three columns
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 btUUfi"
      offset="0"
      size="4"
    >
      One of three columns
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 btUUfi"
      offset="0"
      size="4"
    >
      One of three columns
    </div>
  </div>,
  <div
    class="Row-sc-19nxm1v-0 cwCUyd"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 cPpKgx"
      offset="0"
      size="3"
    >
      One of four columns
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 cPpKgx"
      offset="0"
      size="3"
    >
      One of four columns
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 cPpKgx"
      offset="0"
      size="3"
    >
      One of four columns
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 cPpKgx"
      offset="0"
      size="3"
    >
      One of four columns
    </div>
  </div>,
]
`;

exports[`Grid adaptive 1`] = `
Array [
  <div
    class="Row-sc-19nxm1v-0 cwCUyd"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 iUarAA"
      offset="0"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 bCBXc"
      offset="0"
    >
      2
    </div>
  </div>,
  <div
    class="Row-sc-19nxm1v-0 cwCUyd"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 kVHOpo"
      offset="[object Object]"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 gYgwyj"
      offset="0"
    >
      2
    </div>
  </div>,
]
`;

exports[`Grid w align 1`] = `
Array [
  <div
    class="Row-sc-19nxm1v-0 dMzpbl"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 gYPbBC"
      offset="0"
      size="2"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 gYPbBC"
      offset="0"
      size="2"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 gYPbBC"
      offset="0"
      size="2"
    >
      1
    </div>
  </div>,
  <div
    class="Row-sc-19nxm1v-0 kxHIQF"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 gYPbBC"
      offset="0"
      size="2"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 gYPbBC"
      offset="0"
      size="2"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 gYPbBC"
      offset="0"
      size="2"
    >
      1
    </div>
  </div>,
  <div
    class="Row-sc-19nxm1v-0 ezKqZS"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 gYPbBC"
      offset="0"
      size="2"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 gYPbBC"
      offset="0"
      size="2"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 gYPbBC"
      offset="0"
      size="2"
    >
      1
    </div>
  </div>,
  <div
    class="Row-sc-19nxm1v-0 fBWJak"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 gYPbBC"
      offset="0"
      size="2"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 gYPbBC"
      offset="0"
      size="2"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 gYPbBC"
      offset="0"
      size="2"
    >
      1
    </div>
  </div>,
  <div
    class="Row-sc-19nxm1v-0 eVHAJz"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 gYPbBC"
      offset="0"
      size="2"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 gYPbBC"
      offset="0"
      size="2"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 gYPbBC"
      offset="0"
      size="2"
    >
      1
    </div>
  </div>,
  <div
    class="Row-sc-19nxm1v-0 iobZnc"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 hwTwoH"
      offset="0"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 hwTwoH"
      offset="0"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 hwTwoH"
      offset="0"
    >
      1
    </div>
  </div>,
]
`;

exports[`Grid w valign 1`] = `
Array [
  <div
    class="Row-sc-19nxm1v-0 CXnGS"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 hiyiVh"
      offset="0"
      size="6"
    >
      Tall column
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 hiyiVh"
      offset="0"
      size="6"
    >
      top
    </div>
  </div>,
  <div
    class="Row-sc-19nxm1v-0 dCSdqZ"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 hiyiVh"
      offset="0"
      size="6"
    >
      Tall column
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 hiyiVh"
      offset="0"
      size="6"
    >
      middle
    </div>
  </div>,
  <div
    class="Row-sc-19nxm1v-0 hICVRV"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 hiyiVh"
      offset="0"
      size="6"
    >
      Tall column
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 hiyiVh"
      offset="0"
      size="6"
    >
      bottom
    </div>
  </div>,
  <div
    class="Row-sc-19nxm1v-0 cwCUyd"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 hiyiVh"
      offset="0"
      size="6"
    >
      Tall column
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 hiyiVh"
      offset="0"
      size="6"
    >
      bottom
    </div>
  </div>,
]
`;

exports[`Grid w/ columnGap 1`] = `
Array [
  <div
    class="Row-sc-19nxm1v-0 cwCUyd"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      24
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      24
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      24
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      24
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      24
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      24
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      24
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      24
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      24
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      24
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      24
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      24
    </div>
  </div>,
  <div
    class="Row-sc-19nxm1v-0 eJXEBQ"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      12
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      12
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      12
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      12
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      12
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      12
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      12
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      12
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      12
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      12
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      12
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      12
    </div>
  </div>,
  <div
    class="Row-sc-19nxm1v-0 bwEERC"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      5
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      5
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      5
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      5
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      5
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      5
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      5
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      5
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      5
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      5
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      5
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      5
    </div>
  </div>,
  <div
    class="Row-sc-19nxm1v-0 fhmIwx"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      0
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      0
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      0
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      0
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      0
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      0
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      0
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      0
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      0
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      0
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      0
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 QlXkP"
      offset="0"
      size="1"
    >
      0
    </div>
  </div>,
]
`;

exports[`Grid w/ offset 1`] = `
Array [
  <div
    class="Row-sc-19nxm1v-0 cwCUyd"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 ezcJYy"
      offset="11"
      size="1"
    >
      1
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 fCgaLN"
      offset="10"
      size="2"
    >
      2
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 iNMWJT"
      offset="9"
      size="3"
    >
      3
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 gANdZp"
      offset="8"
      size="4"
    >
      4
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 giUTCw"
      offset="7"
      size="5"
    >
      5
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 cDmPX"
      offset="6"
      size="6"
    >
      6
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 flxgIb"
      offset="5"
      size="7"
    >
      7
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 lgapsP"
      offset="4"
      size="8"
    >
      8
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 dtUlCB"
      offset="3"
      size="9"
    >
      9
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 gLbGIz"
      offset="2"
      size="10"
    >
      10
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 HydGe"
      offset="1"
      size="11"
    >
      11
    </div>
  </div>,
  <div
    class="Row-sc-19nxm1v-0 cwCUyd"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 esTYWy"
      offset="3"
      size="3"
    >
      offset: 3
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 bZmzMl"
      offset="2"
      size="3"
    >
      offset: 2
    </div>
  </div>,
  <div
    class="Row-sc-19nxm1v-0 cwCUyd"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 cPpKgx"
      offset="0"
      size="3"
    >
      size 3
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 bRyXYa"
      offset="3"
      size="2"
    >
      size 2
    </div>
    <div
      class="Colstyles__Box-sc-1nfgix5-0 klLerV"
      offset="3"
      size="1"
    >
      1
    </div>
  </div>,
]
`;

exports[`Grid w/ rowGap 1`] = `
Array [
  <div
    class="Row-sc-19nxm1v-0 iFkyFq"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 cYdJBR"
      offset="0"
      size="12"
    >
      5px row gap
    </div>
  </div>,
  <div
    class="Row-sc-19nxm1v-0 bqTnsj"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 cYdJBR"
      offset="0"
      size="12"
    >
      10px row gap
    </div>
  </div>,
  <div
    class="Row-sc-19nxm1v-0 cwCUyd"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 cYdJBR"
      offset="0"
      size="12"
    >
      No row gap
    </div>
  </div>,
  <div
    class="Row-sc-19nxm1v-0 cwCUyd"
  >
    <div
      class="Colstyles__Box-sc-1nfgix5-0 cYdJBR"
      offset="0"
      size="12"
    >
      No row gap
    </div>
  </div>,
]
`;
